src/EditorFeatures/Core/Implementation/Suggestions/FixMultipleSuggestedAction.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.VisualStudio.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions
+{
+    /// <summary>
+    /// Suggested action for fix multiple occurrences code fix.
+    /// </summary>
+    internal class FixMultipleSuggestedAction : FixAllSuggestedAction
+    {
+        private readonly Document _triggerDocumentOpt;
+        private readonly string _hashOfIds;
+
+        internal FixMultipleSuggestedAction(
+            Workspace workspace,
+            ICodeActionEditHandlerService editHandler,
+            FixMultipleCodeAction codeAction,
+            FixAllProvider provider,
+            ITextBuffer subjectBufferOpt = null)
+            : base(workspace, subjectBufferOpt, editHandler, codeAction, provider, originalFixedDiagnostic: codeAction.GetTriggerDiagnostic())
+        {
+            _triggerDocumentOpt = codeAction.FixAllContext.Document;
+
+            // hash all the diagnostic IDs
+            _hashOfIds = GetHash(codeAction.FixAllContext.DiagnosticIds);
+        }
+
+        private static string GetHash(IEnumerable<string> diagnosticIds)
+        {
+            var hash = 0;
+            foreach (var diagnosticId in diagnosticIds)
```

Comment:
does hash need to be deterministic? or doesn't matter since it is used only once and thrown away?